lib/resty/etcd/v3.lua

```diff
@@ -111,6 +165,16 @@ function _M.new(opts)
     local user       = opts.user
     local password   = opts.password
     local ssl_verify = opts.ssl_verify
+    local failure_window
+    local failure_times
+    local disable_duration
+    local shm_name
+    if opts.health_check then
+        failure_window   = opts.health_check.failure_window
+        failure_times    = opts.health_check.failure_times
+        disable_duration = opts.health_check.disable_duration
+        shm_name         = opts.health_check.shm_name
```

Comment:
need to check if the `shm_name` is a valid share dict name. if it is an invalid name, should throw an error message directly

Reply:
How about this style? It seems clearer.

```lua
local health_check = opts.health_check or {}

failure_window = health_check.failure_window
failure_times = health_check. failure_times
... ...

Reply:
solve it